### ISSUE TITLE
ansible-lint fails to parse alias when anchor is defined in another file

### ISSUE THREAD

**Issue by andrewimeson** [ansible/ansible-lint#4274]
##### Summary

I'm using YAML anchors and aliases to define common connection information in the `main.yml` of my role, and then referencing the anchor with an alias in included task files. Ansible-lint dies with a `load-failure[composererror]: found undefined alias` when parsing the included files

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
$ ansible-lint --version
ansible-lint 24.2.2 using ansible-core:2.15.10 ansible-compat:24.5.1 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
A new release of ansible-lint is available: 24.2.2 → 24.7.0 Upgrade by running: pip install --upgrade ansible-lint
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```yaml
---
# main.yml

- name: Setup FreeIPA Connection and Credential Details
  set_fact:
    ipa_connection: &ipa_connection
      ipa_prot: https
      ipa_host: '{{ lookup("env", "IPA_HOST") | default(inventory_fqdn, true) }}'
      ipa_user: '{{ lookup("env", "IPA_USER") }}'
      ipa_pass: '{{ lookup("env", "IPA_PASS") }}'
      validate_certs: false
    # YAMLFix eats anchors that aren't referenced in the same file, so we'll wastefully
    # duplicate the above structure into a dummy variable.
    __dummy_var:
      <<: *ipa_connection

# Break all configuration into individual files for cleaner organization.
- name: Include configuration snippets
  include_tasks: '{{ item }}'
  with_fileglob:
    - snippet_*.yml
```

and then in an example file

```yaml
---
# snippet_dns_administrator.yml
- name: Create DNS Administrators role group
  community.general.ipa_group:
    name: dns_administrators
    state: present
    <<: *ipa_connection
```

```console (paste below)
$ ansible-lint --exclude=.ansible
WARNING  Skipped installing old role dependencies due to running in offline mode.
WARNING  Skipped installing collection dependencies due to running in offline mode.
WARNING  Listing 1 violation(s) that are fatal
load-failure[composererror]: found undefined alias
  in "<unicode string>", line 11, column 9
roles/freeipa_configuration/tasks/snippet_dns_administrator.yml:1 None


                     Rule Violation Summary
 count tag                         profile rule associated tags
     1 load-failure[composererror] min     core, unskippable

Failed: 1 failure(s), 0 warning(s) on 103 files.
A new release of ansible-lint is available: 24.2.2 → 24.7.0 Upgrade by running: pip install --upgrade ansible-lint
```

##### Desired Behavior

ansible-lint to walk the files the same way Ansible does and load the anchor in one file, and parse the reference in the other.

I suspect this may not be feasible, which is a shame. Does anyone have any other suggestions for avoiding having to duplicate all those connection information on every task?

**Comment by andrewimeson** [comment]
Never mind, this didn't work in Ansible. I think I tested it and forgot to `:w` save my change that split the anchor and alias apart into two files.

My bad.

**Comment by andrewimeson** [comment]
For any future people searching who land here, I found out about `module_defaults` which got me what I needed. I have to define it for each specific FreeIPA module I use, but I can use anchors and aliases in the playbook with no issue.

```yaml
---
# Playbook
- name: Configure global FreeIPA settings from one node
  run_once: true
  hosts:
    - freeipa
  vars:
    ipa_connection: &ipa_connection
      ipa_prot: https
      ipa_host: '{{ lookup("env", "IPA_HOST") | default(inventory_fqdn, true) }}'
      ipa_user: '{{ lookup("env", "IPA_USER") }}'
      ipa_pass: '{{ lookup("env", "IPA_PASS") }}'
      validate_certs: false
  module_defaults:
    community.general.ipa_role:
      <<: *ipa_connection
    community.general.ipa_group:
      <<: *ipa_connection
```

and then those other tasks implicitly pickup the defaults.
